src/services/legal-services.ts

```diff
@@ -182,21 +184,88 @@ export default class LegalServices {
     const url = `businesses/${businessId}/parties`
     return axios.get(url).then(response => {
       const data = response?.data
-      if (!data) throw new Error('Invalid response data')
+      if (!data) throw new Error('Invalid API response')
       return data
     })
   }
 
+  /**
+   * Fetches the directors of the current business.
+   * @param businessId the business identifier
+   * @returns a promise to return the directors from the response, else exception
+   */
+  static async fetchDirectors (businessId: string): Promise<OrgPersonIF[]> {
+    const url = `businesses/${businessId}/directors`
+
+    return axios.get(url)
+      .then(response => {
+        const directors = response?.data?.directors as Array<any>
+
+        if (!directors) throw new Error('Invalid API response')
+
+        // return director list converted to org-person list
+        return directors.map(director => {
+          // WORK-AROUND WARNING !!!
+          // convert directors from "middleInitial" to "middleName"
+          const middleInitial = director.officer['middleInitial']
+          if (middleInitial !== undefined) {
+            director.officer.middleName = middleInitial
+            delete director.officer['middleInitial']
+          }
+
+          const orgPerson: OrgPersonIF = {
+            deliveryAddress: director.deliveryAddress,
+            mailingAddress: director.mailingAddress,
+            officer: director.officer,
+            roles: [
+              {
+                appointmentDate: director.appointmentDate,
+                cessationDate: director.cessationDate,
+                roleType: RoleTypes.DIRECTOR
+              }
+            ],
+            isLookupBusiness: null
+          }
+
+          return orgPerson
+        })
+      })
+  }
+
+  /**
+   * Fetch the share structure of the current business.
+   * @param businessId the business identifier
+   * @returns a promise to return the share structure from the response, else exception
+   */
+  static async fetchShareStructure (businessId: string): Promise<ShareStructureIF> {
+    const url = `businesses/${businessId}/share-classes`
+
+    return axios.get(url)
+      .then(response => {
+        const shareStructure = response.data as ShareStructureIF
+
+        if (!shareStructure) throw new Error('Invalid API response')
+
+        // apply a type to share classes and series
+        shareStructure.shareClasses.forEach(shareClass => {
+          shareClass.type = 'Class'
+          shareClass.series.forEach(shareSeries => { shareSeries.type = 'Series' })
+        })
+
+        return shareStructure
+      })
+  }
+
   /**
    * Fetches addresses.
    * @param businessId the business identifier
    * @returns a promise to return the addresses from the response, else exception
    */
-  static fetchAddresses (businessId: string): Promise<any> {
+  static async fetchAddresses (businessId: string): Promise<any> {
```

Comment:
Again.

I dunno, fetchParties() and THIS were previously awaited, so maybe the code worked correctly even without the async specifier 🤷‍♂️ I just know the previous code was wrong and this is correct.

Reply:
It's weird because I remember my IDE would complain when an await is used on a non-async function. I'm looking now but I don't see my IDE complaining. Anyway, thanks for catching this Sev.